dorado/particle_track.py

```diff
@@ -569,9 +653,73 @@ class (seed_yloc)
             all_walk_data['yinds'] = all_yinds
             all_walk_data['travel_times'] = all_times
 
+            # re-write the walk_data attribute of self
+            self.walk_data = all_walk_data
+
         return all_walk_data
 
 
+def gen_input_check(Np_tracer, seed_xloc, seed_yloc, method):
+    """Check the inputs provided to :obj:`generate_particles()`.
+
+    This function does input type checking and either succeeds or returns
+    an error if the types provided cannot be converted to those that we
+    expect in :obj:`dorado.particle_track.generate_particles()`.
+
+    **Inputs** :
+
+        Np_tracer : `int`
+            Number of particles to generate.
+
+        seed_xloc : `list`
+            List of x-coordinates over which to initially distribute the
+            particles
+
+        seed_yloc : `list`
+            List of y-coordinates over which to initially distribute the
+            particles
+
+        method : `str`, optional
+            Type of particle generation to use, either 'random' or 'exact'.
+
+    **Outputs** :
+
+        Np_tracer : `int`
+            Number of particles to generate.
+
+        seed_xloc : `list`
+            List of x-coordinates over which to initially distribute the
+            particles
+
+        seed_yloc : `list`
+            List of y-coordinates over which to initially distribute the
+            particles
+
+    """
+    if isinstance(Np_tracer, int) is False:
+        try:
+            Np_tracer = int(Np_tracer)
+        except Exception:
+            raise TypeError("Np_tracer input type was not int.")
+    if isinstance(seed_xloc, list) is False:
+        try:
+            seed_xloc = [int(x) for x in list(seed_xloc)]
+        except Exception:
+            raise TypeError("seed_xloc input type was not a list.")
+    if isinstance(seed_yloc, list) is False:
+        try:
+            seed_yloc = [int(y) for y in list(seed_yloc)]
```

Comment:
Since this function will also work for numpy arrays, maybe we should modify the doc strings for `generate_particles` to say that it accepts ndarrays. 

Also, do we need the `isinstance()` check for these `seed` inputs? If not, these same lines could enforce that the indices are type `int` even if the starting format is e.g. a list of floats. If it's too far from a list to work, it'll fail the try and land at the same exception anyway.

Reply:
> Also, do we need the isinstance() check for these seed inputs? If not, these same lines could enforce that the indices are type int even if the starting format is e.g. a list of floats. If it's too far from a list to work, it'll fail the try and land at the same exception anyway.

That's a good point, I guess I can ditch the explicit type check.

Reply:
Oh and regarding changing the doc-string, I would prefer to leave this as a list. We will try to handle flattening arrays and stuff, but the most 'clear' way of passing in something would be as a list I think. Technically the "try" loop would resolve `range()` objects and I'm sure other things as well, but we don't test every exhaustive type that can be converted via `list()` so I don't want to claim that we support them.

Reply:
Oh okay, that makes sense. The fact that it will still work for a variety of inputs is still good news, even if we don't advertise it.